sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -588,7 +584,10 @@ public void setValueDeserializer(String valueDeserializer) {
      * provide the key coder explicitly.
      */
     public Read<K, V> withKeyDeserializer(Class<? extends Deserializer<K>> keyDeserializer) {
-      return toBuilder().setKeyDeserializer(keyDeserializer).build();
+      return toBuilder()
+          .setKeyDeserializer(keyDeserializer)
```

Comment:
I preferred to let it as it was because existing users can rely on `getDeserializer` which would be broken if so. I think we can do this clean up as a follow up after this PR, with the respective deprecation in place.

Reply:
This was to save the state of the object in case someone call the `getDeserializer` method, but it is now inconsistent so probably better to remove it.